### DISCUSSION TITLE
Automated tests

### DISCUSSION THREAD

**Discussion by ljleb** [Mikubill/sd-webui-controlnet#402] [General]
Automated testing, if done right, increases confidence in code changes, allows maintainers and contributors to move faster, often makes it easier to determine the cause of bug reports and helps prevent/detect regressions.

Should we consider adding a test framework to the repo? Maybe some basic code coverage requirements? I'm suggesting this because setting up a bunch of automated eyeballs on some popular code paths may help keep important use-cases in check.

**Comment by Mikubill** [comment]
I agree, maybe we can (also) add something like https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/.github/workflows/run_tests.yaml after #384 and #393 get merged
